LedgerSMB/PGDate.pm

```diff
@@ -148,10 +149,11 @@ sub from_input{
     $format = 'yyyy-mm-dd' if $input =~ /^\d{4}/;
     my $dt =  _parse_string($self, $input, uc($format), $has_time)
 		  if $input;
-    my $rv = $self->now();
-    $rv->{date} = $dt
-        if $dt;
-    return $rv;
+    my $retval = $self->new($ddt);
```

Comment:
@einhverfr on the line above, the variable $dt is being assigned; on this line, you use the variable $ddt; a bug?